code/datums/martial/boxing.dm

```diff
@@ -1,102 +1,273 @@
+#define LEFT_RIGHT_COMBO "DH"
+#define RIGHT_LEFT_COMBO "HD"
+#define LEFT_LEFT_COMBO "HH"
+#define RIGHT_RIGHT_COMBO "DD"
+
 /datum/martial_art/boxing
 	name = "Boxing"
 	id = MARTIALART_BOXING
 	pacifist_style = TRUE
+	///Boolean on whether we are sportsmanlike in our tussling; TRUE means we have restrictions
+	var/honorable_boxer = TRUE
+	/// List of traits applied to users of this martial art.
+	var/list/boxing_traits = list(TRAIT_BOXING_READY)
 
 /datum/martial_art/boxing/teach(mob/living/new_holder, make_temporary)
 	if(!ishuman(new_holder))
 		return FALSE
+	new_holder.add_traits(boxing_traits, BOXING_TRAIT)
+	RegisterSignal(new_holder, COMSIG_LIVING_CHECK_BLOCK, PROC_REF(check_block))
+	return ..()
+
+/datum/martial_art/boxing/on_remove(mob/living/remove_from)
+	remove_from.remove_traits(boxing_traits, BOXING_TRAIT)
+	UnregisterSignal(remove_from, list(COMSIG_LIVING_CHECK_BLOCK))
 	return ..()
 
-/datum/martial_art/boxing/disarm_act(mob/living/carbon/human/attacker, mob/living/defender)
-	attacker.balloon_alert(attacker, "can't disarm while boxing!")
-	return MARTIAL_ATTACK_FAIL
+///Unlike most instances of this proc, this is actually called in _proc/tussle()
+///Returns a multiplier on our skill damage bonus.
+/datum/martial_art/boxing/proc/check_streak(mob/living/attacker, mob/living/defender)
+	var/combo_multiplier = 1
+	
+	if(findtext(streak, LEFT_LEFT_COMBO) || findtext(streak, RIGHT_RIGHT_COMBO))
+		reset_streak()
+		attacker.balloon_alert(attacker, "weak combo, alternate your hits!")
```

Comment:
Maybe a little long for a balloon alert you can trigger every click, although to be honest I don't have a better suggestion either

Reply:
Its thankfully every other click actually, so it shouldn't be too frequent and it didn't seem to pop up too much.